ledger/src/shred/merkle.rs

```diff
@@ -148,12 +150,14 @@ impl ShredData {
     //   ShredCode::capacity(proof_size).unwrap()
     //       - ShredData::SIZE_OF_HEADERS
```

Comment:
I see that `ShredCode::capacity` already subtracts `ShredCode::SIZE_OF_HEADERS`. Should this be
```
- (ShredData::SIZE_OF_HEADERS - ShredCode::SIZE_OF_HEADERS)
```

Reply:
I think this is correct as is, and is described why here:
https://github.com/solana-labs/solana/blob/8ad125d0c/ledger/src/shred.rs#L5-L50

We also have a test for this:
https://github.com/solana-labs/solana/blob/8ad125d0c/ledger/src/shred/merkle.rs#L1338-L1347

Reply:
got it, so since the data header is also erasure coded and data shred must be the same length of coding shred, we add the restricted section, which is the size of the coding shred header.